internal/parse/parser.go

```diff
@@ -243,58 +236,62 @@ func (p *Parser) skipName() bool {
 //  - bool == false, err == nil
 //		The construct was not the one we are looking for
 
-// parseIdentifier parses either a name made up only of nameBytes or an
-// asterisk.
-func (p *Parser) parseIdentifier() (string, bool) {
+// parseIdentifierAsterisk parses a name made up of only nameBytes or of a
+// single asterisk. On success it returns the parsed string and true. Otherwise,
+// it returns the empty string and false.
+func (p *Parser) parseIdentifierAsterisk() (string, bool) {
 	if p.skipByte('*') {
 		return "*", true
 	}
+	return p.parseIdentifier()
+}
 
-	idStart := p.pos
+// parseIdentifier parses a name made up of only nameBytes. On success it
+// returns the parsed string and true. Otherwise, it returns the empty string
+// and false.
+func (p *Parser) parseIdentifier() (string, bool) {
+	mark := p.pos
 	if p.skipName() {
-		return p.input[idStart:p.pos], true
+		return p.input[mark:p.pos], true
 	}
 	return "", false
 }
 
-// parseFullName parses a column name or Go field name, optionally dot-prefixed by
-// its table or type name respectively.
-func (p *Parser) parseFullName(idc idClass) (FullName, bool) {
+// parseGoTarget parses a go type name qualified by a tag name (or asterisk) of
+// the form "TypeName.col_name". On success it returns the parsed FullName, true
+// and nil. If a target is found, but not formatted correctly, false and an
+// error are returned. Otherwise the error is nil.
+func (p *Parser) parseGoTarget() (FullName, bool, error) {
```

Comment:
Isn't this both Source and Target?  If so, parseGoFullName might be more appropriate. If not, then it's fine.

Reply:
Changed.